src/modules/FacilitesModule/components/FacilitesList/FacilitesList.tsx

```diff
@@ -1,8 +1,435 @@
+import Paper from "@mui/material/Paper";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell, { tableCellClasses } from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import { TextField } from "@mui/material";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+import DialogContent from "@mui/material/DialogContent";
+import DialogActions from "@mui/material/DialogActions";
 
+import { useForm } from "react-hook-form";
+
+import { Edit } from "@mui/icons-material";
+import CloseIcon from "@mui/icons-material/Close";
+
+import MoreVertIcon from "@mui/icons-material/MoreVert";
+import {
+  Backdrop,
+  Box,
+  Button,
+  Grid,
+  IconButton,
+  Menu,
+  MenuItem,
+  Modal,
+  Stack,
+  TablePagination,
+  Typography,
+
+} from "@mui/material";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+
+import DeleteIcon from "@mui/icons-material/Delete";
+import { axiosInstanceWithHeaders } from "../../../../axiosConfig/axiosInstance";
+import DeleteData from "../../../SharedModule/components/DeleteData/DeleteData";
+import Loading from "../../../SharedModule/components/Loading/Loading";
+import { toast } from "react-toastify";
+import { Facilities } from "../../../../interfaces/interface";
+
+const StyledTableCell = styled(TableCell)(({ theme }) => ({
+  [`&.${tableCellClasses.head}`]: {
+    color: theme.palette.common.black,
+    fontWeight: "bold",
+  },
+  [`&.${tableCellClasses.body}`]: {
+    fontSize: 14,
+  },
+}));
+
+const StyledTableRow = styled(TableRow)(({ theme }) => ({
+  "&:nth-of-type(odd)": {
+    backgroundColor: theme.palette.action.hover,
+  },
+  "&:last-child td, &:last-child th": {
+    border: 0,
+  },
+}));
+
+const ActionTableCell = styled(TableCell)({
+  textAlign: "right",
+  p: 3,
+});
+
+
+
+
+// /////////////////////style Model
+const BootstrapDialog = styled(Dialog)(({ theme }) => ({
```

Comment:
This component is used in multiple pages. We can make it in one place and then import it here and in any place we want to use.